<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>Yingxin Lin - Publications</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 20,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit"
  }
}</script>


<link rel="stylesheet" href="styles.css">
<meta property="og:title" content="Yingxin Lin - Publications">
<meta property="og:description" content="†: Co-first/co-last author">
<meta property="og:image" content="img/icon.png">
<meta property="og:site-name" content="Yingxin Lin">
<meta name="twitter:title" content="Yingxin Lin - Publications">
<meta name="twitter:description" content="†: Co-first/co-last author">
<meta name="twitter:creator" content="@LinYingxin">
<meta name="twitter:site" content="@LinYingxin">
<meta name="twitter:card" content="summary_large_image">
</head>

<body class="floating nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg navbar-dark ">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container">
    <a class="navbar-brand" href="./index.html">
    <span class="navbar-title">Yingxin Lin</span>
    </a>
  </div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="./index.html">
 <span class="menu-text">Home</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link active" href="./publication.html" aria-current="page">
 <span class="menu-text">Publications</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./projects.html">
 <span class="menu-text">Research</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./softwares.html">
 <span class="menu-text">Softwares</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="./talk.html">
 <span class="menu-text">Talks</span></a>
  </li>  
</ul>
              <div id="quarto-search" class="" title="Search"></div>
          </div> <!-- /navcollapse -->
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<header id="title-block-header" class="quarto-title-block default toc-left page-columns page-full">
  <div class="quarto-title-banner page-columns page-full">
    <div class="quarto-title column-body">
      <h1 class="title">Publications</h1>
                      </div>
  </div>
    
  
  <div class="quarto-title-meta">

      
    
      
    </div>
    
  
  </header><div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse sidebar-navigation floating overflow-auto">
    <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Publication categories</h2>
   
  <ul>
  <li><a href="#preprint" id="toc-preprint" class="nav-link active" data-scroll-target="#preprint">Preprint</a></li>
  <li><a href="#publication" id="toc-publication" class="nav-link" data-scroll-target="#publication">Publication</a>
  <ul>
  <li><a href="#methodologies-and-tools" id="toc-methodologies-and-tools" class="nav-link" data-scroll-target="#methodologies-and-tools">Methodologies and tools</a></li>
  <li><a href="#applications" id="toc-applications" class="nav-link" data-scroll-target="#applications">Applications</a>
  <ul class="collapse">
  <li><a href="#single-cell" id="toc-single-cell" class="nav-link" data-scroll-target="#single-cell">Single-cell</a></li>
  <li><a href="#other-omics" id="toc-other-omics" class="nav-link" data-scroll-target="#other-omics">Other omics</a></li>
  <li><a href="#kidney-transplant" id="toc-kidney-transplant" class="nav-link" data-scroll-target="#kidney-transplant">Kidney transplant</a></li>
  </ul></li>
  <li><a href="#review-news" id="toc-review-news" class="nav-link" data-scroll-target="#review-news">Review &amp; news</a></li>
  </ul></li>
  </ul>
</nav>
</nav>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
    </div>
<!-- main -->
<main class="content quarto-banner-title-block" id="quarto-document-content">




<p>†: Co-first/co-last author</p>
<section id="preprint" class="level2">
<h2 class="anchored" data-anchor-id="preprint">Preprint</h2>
<ul>
<li><p>Zhang, D. †, Rubio Rodríguez-Kirby, L. †, <strong>Lin, Y.</strong>, Song, M., Wang, L., …, &amp; Zhao, H., Castelo-Branco, G., Fan, R. (2024) Spatial dynamics of mammalian brain development and neuroinflammation by multimodal tri-omics mapping. [<a href="https://doi.org/10.1101/2024.07.28.605493">paper</a>]</p></li>
<li><p>Lim, S.Y., <strong>Lin, Y.</strong>, Lee, J.H., Pedersen, B., Stewart, A., Scolyer, R.A., Long, G.V., Yang, J.Y.H., Rizos, H. (2024) Single-cell RNA sequencing reveals melanoma cell state-dependent heterogeneity of response to MAPK inhibitors. [<a href="https://dx.doi.org/10.2139/ssrn.4805810">paper</a>]</p></li>
<li><p>Hsu, I. U., <strong>Lin, Y.</strong>, Guo, Y., Xu, Q. J., Shao, Y., Wang, R. L., Yin, D., Zhao, J., Young, L.H., Zhao, H., Zhang, L., Chang, R. B. (2023) Differential developmental blueprints of organ-intrinsic nervous systems. [<a href="https://doi.org/10.1101/2023.12.12.571306">paper</a>]</p></li>
<li><p>Cao, Y., Yu, L., Torkel, M., Kim, S., <strong>Lin, Y.</strong>, Yang, P., Speed, T., Ghazanfar, S., Yang, J.Y.H. (2023) The current landscape and emerging challenges of benchmarking single-cell methods. [<a href="https://doi.org/10.1101/2023.12.19.572303">paper</a>]</p></li>
<li><p>Yin, D., Cao, Y., Chen, J., Mak, C.L.Yl, Yu, K.H.O., <strong>Lin, Y.</strong> †, Ho, J. W. K. †, Yang, J.Y.H. †. (2022) Covidscope: An atlas-scale COVID-19 resource for single-cell meta analysis at sample and cell levels. [<a href="https://doi.org/10.1101/2022.12.03.518997">paper</a>]</p></li>
</ul>
</section>
<section id="publication" class="level2">
<h2 class="anchored" data-anchor-id="publication">Publication</h2>
<section id="methodologies-and-tools" class="level3">
<h3 class="anchored" data-anchor-id="methodologies-and-tools">Methodologies and tools</h3>
<ul>
<li><p>Li, H. †, <strong>Lin, Y.</strong>†, He, W., Han, W., Xu, X., Xu, C., Gao, E., Zhao, H., Gao, X. (2024) SANTO: a coarse-to-fine stitching and alignment method for spatial omics. <strong>Nature Communications</strong> [<a href="https://doi.org/10.1038/s41467-024-50308-x">paper</a>]</p></li>
<li><p>Fu, X.†, <strong>Lin, Y.</strong>†, Lin, D., Mechtersheimer, D., Wang, C., Ameen, F., Ghazanfar, S., Patrick, E., Kim, J. and Yang, J.Y.H. (2024) Biologically-informed self-supervised learning for segmentation of subcellular spatial transcriptomics data. <strong>Nature Communications</strong> [<a href="https://doi.org/10.1038/s41467-023-44560-w">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>†, Wu, T.Y.†, Chen, X.†, Wan, S., Chao, B., Xin, J., Yang, J.Y.H., Wong, W.H. and Wang, Y.R. (2024) Data integration and inference of gene regulation using single-cell temporal multimodal data with scTIE. <strong>Genome Research</strong> [<a href="https://doi.org/10.1101/gr.277960.123">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>, Cao, Y., Willie, E., Patrick, E., Yang, J.Y.H. (2023) Atlas-scale single-cell multi-sample multi-condition data integration using scMerge2. <strong>Nature Communications</strong> [<a href="https://doi.org/10.1038/s41467-023-39923-2">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>†, Wu, T.Y.†, Wan, S., Yang, J.Y.H., Wong, W.H., and Wang, Y.X.R.(2022) scJoint: transfer learning for data integration of single-cell RNA-seq and ATAC-seq. <strong>Nature Biotechnology.</strong> [<a href="https://github.com/SydneyBioX/scJoint">package</a>][<a href="https://doi.org/10.1038/s41587-021-01161-6">paper</a>]</p></li>
<li><p>Cao, Y., <strong>Lin, Y.</strong>, Patrick, E., Yang, P., and Yang, J.Y.H. (2022) scFeatures: Multi-view representations of single-cell and spatial data for disease outcome prediction. <strong>Bioinformatics</strong>. [<a href="https://github.com/SydneyBioX/scFeatures">package</a>][<a href="https://doi.org/10.1093/bioinformatics/btac590">paper</a>]</p></li>
<li><p>Kim, H.J., Wang, K.Y., Chen, C., <strong>Lin, Y.</strong>, Tam, P., Lin, D.M., Yang, J.Y.H. and Yang, P. (2021) Uncovering cell identity through differential stability with Cepo. <strong>Nature Computational Science</strong>. [<a href="https://PYangLab.github.io/Cepo">package</a>][<a href="https://doi.org/10.1038/s43588-021-00172-2">paper</a>]</p></li>
<li><p>Ghazanfar, S., <strong>Lin, Y.</strong>, Su, X., Lin, D.M., Patrick, E., Han, Z.G., Marioni, J.C., and Yang, J.Y.H. (2020) Investigating higher-order interactions in single-cell data with scHOT. <strong>Nature Methods</strong>. [<a href="https://shazanfar.github.io/scHOT">package</a>][<a href="https://doi.org/10.1038/s41592-020-0885-x">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>, Cao, Y., Kim, H.J., Salim, A., Speed, T., Lin, D.M., Yang, P. and Yang, J.Y.H. (2020) scClassify: sample size estimation and multiscale classification of cells using single and multiple reference. <strong>Molecular Systems Biology</strong>. [<a href="https://sydneybiox.github.io/scClassify">package</a>][<a href="http://shiny.maths.usyd.edu.au/scClassify">shiny</a>][<a href="https://doi.org/10.15252/msb.20199389">paper</a>] </p></li>
<li><p>Kim, H.J.†, <strong>Lin, Y.†</strong>, Geddes, T.A., Yang, J.Y.H. and Yang, P. (2020) CiteFuse enables multi-modal analysis of CITE-seq data. <strong>Bioinformatics</strong>. [<a href="https://sydneybiox.github.io/CiteFuse">package</a>][<a href="http://shiny.maths.usyd.edu.au/CiteFuse">shiny</a>][<a href="https://doi.org/10.1093/bioinformatics/btaa282">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>, Ghazanfar, S., Wang, K.Y., Gagnon-Bartsch, J.A., Lo, K.K., Su, X., Han, Z.G., Ormerod, J.T., Speed, T.P., Yang, P. and Yang, J.Y.H. (2019) scMerge leverages factor analysis, stable expression, and pseudoreplication to merge multiple single-cell RNA-seq datasets. <strong>Proceedings of the National Academy of Sciences</strong>. [<a href="https://sydneybiox.github.io/scMerge">package</a>] [<a href="https://doi.org/10.1073/pnas.1820006116">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>, Ghazanfar, S., Strbenac, D., Wang, A., Patrick, E., Lin, D., Speed, T., Yang, J.Y.H. &amp; Yang, P. (2019) Evaluating stably expressed genes in single cells. <strong>GigaScience</strong>. [<a href="https://github.com/SydneyBioX/scMerge">package</a>] [<a href="https://doi.org/10.1093/gigascience/giz106">paper</a>]</p></li>
<li><p>Cao, Y.†, <strong>Lin, Y.</strong>†, Ormerod, J., Yang, P., Yang, J.Y.H. &amp; Lo, K. (2019) scDC: single cell differential composition analysis. <strong>BMC Bioinformatics</strong>. [<a href="https://sydneybiox.github.io/scDC">package</a>] [<a href="https://doi.org/10.1186/s12859-019-3211-9">paper</a>]</p></li>
<li><p>Kim, T., Chen, I.R., <strong>Lin, Y.</strong>, Wang, A.Y.Y., Yang, J.Y.H. and Yang, P. (2019) Impact of similarity metrics on single-cell RNA-seq data clustering. <strong>Briefings in bioinformatics</strong>, 20(6), pp.2316-2326. [<a href="https://doi.org/10.1186/s12859-019-3211-9">paper</a>]</p></li>
</ul>
</section>
<section id="applications" class="level3">
<h3 class="anchored" data-anchor-id="applications">Applications</h3>
<section id="single-cell" class="level4">
<h4 class="anchored" data-anchor-id="single-cell">Single-cell</h4>
<ul>
<li>Cao, Y., Tran, A., Kim, H., Robertson, N., <strong>Lin, Y.</strong>, Torkel, M., Yang, P., Patrick, E., Ghazanfar, S., Yang, J.Y.H. (2023) Thinking process templates for constructing data stories with SCDNEY. [<a href="https://f1000research.com/articles/12-261">paper</a>]</li>
<li><strong>Lin, Y.</strong>, Loo, L., Tran, A., Moreno, C., Hesselson D., Neely, G., and Yang, J.Y.H. (2022) Characterization of cell-cell communication in COVID-19 patients. <strong>PLoS Computational Biology</strong>. [<a href="https://doi.org/10.1101/2020.12.30.424641">paper</a>]</li>
<li>He, J.†, <strong>Lin, Y.</strong>†, Meng M., Li J., Yang, J.Y.H., Wang H. (2021) Construction of a human cell landscape of COVID-19 infection at single-cell level. <strong>Aging and disease</strong>. [<a href="https://doi.org/10.14336%2FAD.2021.0301">paper</a>]</li>
<li>Su, X., Zhao, L., Shi, Y., Zhang, R., Long, Q., Bai, S., Luo, Q., <strong>Lin, Y.</strong>, Zou, X., Ghazanfar, S. and Tao, K., … &amp; Han, Z.G. (2021) Clonal evolution in liver cancer at single-cell and single-variant resolution. <strong>Journal of hematology &amp; oncology</strong>.[<a href="https://doi.org/10.1186/s13045-021-01036-y">paper</a>]</li>
<li>Su, X., Long, Q., Bo, J., Shi, Y., Zhao, L.N., <strong>Lin Y.,</strong> Luo, Q., Ghazanfar, S., Zhang, C., Liu, Q., Wang, L., He, K., He, J., Cui, X., Yang, J.Y.H., Han, Z.G., Yang, G., and Sha, J.J. (2020) Mutational and transcriptomic landscapes of a rare human prostate basal cell carcinoma. <strong>The Prostate</strong>. [<a href="https://doi.org/10.1002/pros.23965">paper</a>]</li>
</ul>
</section>
<section id="other-omics" class="level4">
<h4 class="anchored" data-anchor-id="other-omics">Other omics</h4>
<ul>
<li>Martinez, A. P., Chung, A. C., Huang, S., Bisogni, A. J., <strong>Lin, Y.</strong>, Cao, Y., … &amp; Lin, D. M. (2023) Pcdh19 mediates olfactory sensory neuron coalescence during postnatal stages and regeneration. <strong>Iscience</strong>, 26(11).</li>
</ul>
</section>
<section id="kidney-transplant" class="level4">
<h4 class="anchored" data-anchor-id="kidney-transplant">Kidney transplant</h4>
<ul>
<li><p>Philipoff, A. †, <strong>Lin, Y.</strong>†, Teixeira-Pinto, A., Gately, R., Craig, J.C., …, Wong, G. &amp; Lim, W. (2024) Antecedent Cardiac Arrest Status of Donation After Circulatory Determination of Death (DCDD) Kidney Donors and the Risk of Delayed Graft Function After Kidney Transplantation: A Cohort Study. <strong>Transplantation</strong>. [<a href="https://doi.org/10.1097/tp.0000000000005022">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>, Teixeira-Pinto, A., Craig, J., Opdam, H., Chapman, J., Pleass, H., Carter, A., Rogers N., Davies, C., McDonald, S., Yang, J., Lim, W., and Wong, G. (2023) Trajectories of systolic blood pressure decline in kidney transplant donors prior to circulatory death and delayed graft function. <strong>Clinical Kidney Journal</strong>. [<a href="https://doi.org/10.1093/ckj/sfad047">paper</a>]</p></li>
<li><p>Zhang, Y., Hu, A., <strong>Lin, Y.</strong>, Cao, Y., Muller, S., Wong, G., and Yang, J.Y.H. (2023) simKAP: Simulation framework for the kidney allocation process with decision making model. <strong>Scientific Reports</strong> [<a href="https://doi.org/10.1038/s41598-023-41162-w">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong>, Teixeira-Pinto, A., Opdam, H., Chapman, J. R., Craig, J. C., Rogers, N., … &amp; Wong, G. (2022) Nonutilization of Kidneys From Donors After Circulatory Determinant of Death. <strong>Transplantation Direct</strong>. [<a href="https://doi.org/10.1097%2FTXD.0000000000001331">paper</a>]</p></li>
<li><p>Lim, W.H., Ooi E., Pankaj, A., Teixeira-Pinto, A., <strong>Lin, Y.</strong>, … &amp; Wong, G. (2022) The incidence of cancer recurrence and new cancer following commencement of dialysis. <strong>Clinical Kidney Journal</strong>. [<a href="https://doi.org/10.1093/ckj/sfac124">paper</a>]</p></li>
<li><p>Chung, E., Blazek, K., Teixeira-Pinto, A., Sharma, A., Kim, S., <strong>Lin, Y.</strong>, … &amp; Wong, G. (2022) Predictive Models for Recurrent Membranous Nephropathy After Kidney Transplantation. <strong>Transplantation Direct</strong>. [<a href="https://doi.org/10.1097/TXD.0000000000001357">paper</a>]</p></li>
</ul>
</section>
</section>
<section id="review-news" class="level3">
<h3 class="anchored" data-anchor-id="review-news">Review &amp; news</h3>
<ul>
<li><p>Kim, D., Tran, A., Kim, H.J., <strong>Lin, Y.</strong>, Yang, J.Y.H. and Yang, P. (2023) Gene regulatory network reconstruction: harnessing the power of single-cell multi-omic data. <strong>npj Systems Biology and Applications</strong>. [<a href="https://doi.org/10.1038/s41540-023-00312-6">paper</a>]</p></li>
<li><p><strong>Lin, Y.</strong> and Yang, J.Y. (2022) 3D reconstruction of spatial expression. <strong>Nature Methods</strong>. [<a href="https://doi.org/10.1038/s41592-022-01476-5">paper</a>]</p></li>
</ul>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">© 2024 Yingxin Lin</div>   
    <div class="nav-footer-right">
      <ul class="footer-items list-unstyled">
    <li class="nav-item compact">
    <a class="nav-link" href="https://github.com/yingxinlin">
      <i class="bi bi-github" role="img">
</i> 
    </a>
  </li>  
    <li class="nav-item compact">
    <a class="nav-link" href="mailto:yingxin.lin@yale.edu">
      <i class="bi bi-envelope" role="img">
</i> 
    </a>
  </li>  
</ul>
    </div>
  </div>
</footer>



</body></html>