---
output:
  xaringan::moon_reader:
    css: ["default", "assets/sydney-fonts.css", "assets/sydney.css"]
    self_contained: false # if true, fonts will be stored locally
    seal: true # show a title slide with YAML information
    includes:
      in_header: "assets/mathjax-equation-numbers.html"
    nature:
      beforeInit: ["assets/remark-zoom.js", "https://platform.twitter.com/widgets.js"]
      highlightStyle: github
      highlightLines: true
      countIncrementalSlides: false
      ratio: '16:9' # alternatives '16:9' or '4:3' or others e.g. 13:9
      navigation:
        scroll: false # disable slide transitions by scrolling
        
---


```{r, load_refs, echo=FALSE, cache=FALSE}
library(RefManageR)
BibOptions(check.entries = FALSE, 
           bib.style = "authoryear", 
           cite.style = 'authoryear', 
           style = "markdown",
           hyperlink = FALSE, 
           dashed = FALSE)
myBib <- ReadBib("assets/example.bib", check = FALSE)
```


class: title-slide
background-image: url("assets/USydLogo-black.svg"), url("assets/title-image3.jpg")
background-position: 10% 90%, 100% 50%
background-size: 160px, 50% 100%
background-color: #E64626

.pull-left[
# .black[scMerge]

## Integration of multiple single-cell RNA-seq datasets leveraging stable expression and pseudo-replication

### Yingxin Lin

### Nov 30, 2018



<br><br><br>

]

---
class: segue-red

# scMerge algorithm




---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide8.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide9.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide10.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide11.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide12.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide13.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide14.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide15.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide16.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide17.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide18.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide19.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide20.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide21.png")
background-size: contain
background-position: 50% 50%

---
class: segue-red

# Evaluation




---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide31.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/sil.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/sil2.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide32.png")
background-size: contain
background-position: 50% 50%

---



## Installing `r icon::fa("download",pull = "right")`

The **scMerge** package is in GitHub code repository [`r icon::fa("github")` `SydneyBioX/scMerge`](https://github.com/SydneyBioX/scMerge)

```{r, eval = FALSE}
# Installing scMerge and the data files using
devtools::install_github("SydneyBioX/scMerge.data")
devtools::install_github("SydneyBioX/scMerge")
```

To load the `scMerge` package. We designed our package to be consistent with the popular BioConductorâ€™s single cell analysis framework, namely the `SingleCellExperiment` package and `scater`.


```{r message = FALSE}
library(scMerge.data)
library(scMerge)
library(SingleCellExperiment)
library(scater)
```


---


class: segue-red


# Case Study

Unsupervised scMerge

Supervised scMerge

Semi-supervised scMerge

---

## ESC data

We will use a mouse embryonic stem cell (ESC) data as an example to illustrate  `scMerge` function. 

.scroll-output[
```{r}
## mouse ESC data
data("sce_mESC", package = "scMerge.data")
sce_mESC
```
]

---

## ESC data

The mESC data contains 5 different batches from three different cell types. Using a PCA plot, we can see that despite strong separation of cell types, there is also a strong separation due to batch effects. 


.pull-left[
```{r}
table(sce_mESC$batch, sce_mESC$cellTypes)
```
]

.pull-right[
```{r fig.height=4, fig.width=5, cahce = T}
scater::plotPCA(sce_mESC, 
                colour_by = "cellTypes", 
                shape_by = "batch")
```
]



---
## ESC data: running unsupervised `scMerge`



Key inputs of `scMerge`:

- `sce_combine`: a `SingleCellExperiment` object contains the batch-combined matrix with batch info in `colData` and `counts` and `logcounts` matrices in `assay` slot.
- `ctl`: a gene list of negative control genes.
- `assay_name`: The assay name for the adjusted expression matrix. 

--

In unsupervised scMerge, we will perform a k-means clustering to obtain pseudo-replicates. This requires the users to supply a `kmeansK` vector with each element indicating number of clusters in each of the batches. 

--

For example, we know the first, forth, and fifth batch contains only one cell type and the second and third batch contains three cell types. Hence, `kmeansK = c(1,3,3,1,1)` in this case.

--

```{r results = "hide", fig.show = 'hide', cache = T}
sce_mESC <- scMerge(sce_combine = sce_mESC, 
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_unsupervised")
```

---
## ESC data: running unsupervised `scMerge`

```{r results = "hide", fig.show = 'hide', cache = F, eval = F}
sce_mESC <- scMerge(sce_combine = sce_mESC, 
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_unsupervised")
```

The normalised data matrix will be stored in slot `assays` with the list name as `"scMerge_unsupervised"` of `sce_mESC`.

--

.blockquote[
### `r icon::fa("comment-dots")` Comments

scMerge is robust to overestimation of number of clusters in general (See our manuscript for details).
]


---

## ESC data: running unsupervised `scMerge`

We now construct the PCA plot again on our normalised data. We can observe a much better separation by cell type and less separation by batches.

```{r fig.height=5, fig.width=6, cache = T}
scater::plotPCA(sce_mESC, 
                colour_by = "cellTypes", 
                shape_by = "batch",
                run_args = list(exprs = "scMerge_unsupervised"))
```


---

## Other diagnostic plots

**Explanatory variables plot**: density plot of percentage of phenotypic variance explained.



```{r echo = F, message=FALSE, warning=FALSE, results='hide'}
sce_mESC <- calculateQCMetrics(sce_mESC)
```


.pull-left[
```{r fig.height=4, fig.width=5, cache = T}
scater::plotExplanatoryVariables(sce_mESC,
        method = "density",
        exprs_values = "logcounts", 
        variables = c("batch", "cellTypes",
                      "total_counts",
                      "total_features")) + 
  labs(title = "logcounts")
```
]

.pull-right[
```{r fig.height=4, fig.width=5, cache = T}
scater::plotExplanatoryVariables(sce_mESC,
        method = "density",
        exprs_values = "scMerge_unsupervised", 
        variables = c("batch", "cellTypes",
                      "total_counts",
                      "total_features")) + 
  labs(title = "scMerge")
```
]

---

## ESC data: running supervised `scMerge`

If **all** cell type information is available to the users, or users has already performed clustering analysis using their favourite method, then it is possible to use this information to create pseudo-replicates. 

This can be done through the `cell_type` argument in the `scMerge` function.


```{r eval = FALSE}
sce_mESC <- scMerge(sce_mESC,
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_supervised",
                  {{cell_type = sce_mESC$cellTypes}})
```

---

## ESC data: running semi-supervised `scMerge` 

If the user is only able to access **partial** cell type information (or have confidence in only partial cell type informaton), then it is still possible to use this information to create pseudo-replicates. 

This can be done through the `cell_type` and `cell_type_inc` arguments in the scMerge function. `cell_type_inc` should contain a vector of indices indicating which elements in the `cell_type` vector should be used to perform semi-supervised `scMerge`.


```{r eval = FALSE}
sce_mESC <- scMerge(sce_mESC,
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_semisupervised1",
                    cell_type = sce_mESC$cellTypes,
                  {{cell_type_inc = which(sce_mESC$cellTypes == "2i"}}))
```

---


## ESC data: running semi-supervised `scMerge` 
There is alternative semi-supervised method to create pseudo-replicates for scMerge. This uses known cell type information to identify mutual nearest clusters.
It is achieved via the `cell_type` and `cell_type_match = TRUE` options in the `scMerge` function.


```{r eval = FALSE}
sce_mESC <- scMerge(sce_mESC,
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_semisupervised2",
                    cell_type = sce_mESC$cellTypes,
                    {{cell_type_match = TRUE}})
```


---

## Achieving fast computation: `fastRUVIII` `r icon::fa("rocket",pull = "right")`


`RUVIII` function in the package `ruv` is great but slow for the large single-cell data. We develop a fast version of scMerge, by using 
- *randomised SVD algorithm*: to obtain a very accurate approximation of the noise structure in the data by performing a partial SVD. 
-  package `Rcpp` to speed up the matrix calculation.
- `rsvd_prop` is a parameter between 0 and 1, controlling the degree of approximations.

.pull-left[
```{r eval = FALSE}
fastRUVIII(Y, M, ctl, k = NULL, fast_svd = TRUE,
           rsvd_prop = 0.1)
```

```{r eval = F}
sce_mESC <- scMerge(sce_mESC, 
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_fast", 
                   {{fast_svd = TRUE, 
                   rsvd_prop = 0.1}})
```

]

--

.pull-right[
.blockquote[
### `r icon::fa("comment-dots")` Comments

We recommend only use this option when the number of cells is large in your single cell data. 
]]


---


background-image: url(assets/scMergeSlides/speedup.png)
background-size: 550px
background-position: 20% 80%

## Achieving fast computation: `fastRUVIII` `r icon::fa("rocket",pull = "right")`

For the case with 10k cell, the speed up of `fastRUVIII` is about 10 times, compared to `RUVIII` from package `ruv`. 
---

<!-- ## Achieving fast computation: `BiocParallel` `r icon::fa("rocket",pull = "right")` -->

<!-- We also utilise `BiocParallel` package to parallise the identification of pseudo-replicates. -->

<!-- ```{r eval = F} -->
<!-- sce_mESC <- scMerge(sce_mESC,  -->
<!--                     ctl = segList_ensemblGeneID$mouse$mouse_scSEG, -->
<!--                     kmeansK = c(1,3,3,1,1), -->
<!--                     assay_name = "scMerge_fast",  -->
<!--                   {{parallel = TRUE, -->
<!--                   parallelParam = bpparam()}}) -->
<!-- ``` -->




---

## More on stably expressed genes

### Other negative controls

`scMerge` also provides some other gene lists of negative control genes.

```{r}
data("segList")
str(segList)
```

---

## More on stably expressed genes


### Alternatively, you define your own stably expressed gene list

`scSEGIndex()` will calculate the single-cell stably expressed gene index to identify stably expressed genes (SEGs) from a log transformed single-cell data. The data here is assumed to be no batch effect and covered a wide range of cell types.

```{r eval = FALSE}
scSEGIndex(exprsMat, cell_type = NULL, ncore = 1)
```






---

## Reference list

```{r echo = FALSE, results = "hide"}
Citet(myBib,"lin2018scmerge")
Citet(myBib,"lin2017housekeeping")
Citet(myBib,"ghazanfar2016integrated")
```



```{r, results = "asis", echo=FALSE}
PrintBibliography(myBib)
```



---

background-image: url(assets/USydLogo-black.svg)
background-size: 260px
background-position: 5% 95%

## Acknowledgement


.pull-left[
### The University of Sydney
- Jean Yang
- Pengyi Yang
- John Ormerod
- Shila Ghazanfar
- Kevin Wang
- Kitty Lo
]


.pull-right[
### WEHI
- Terry Speed

### The University of Michigan
- Johann Gagnon-Bartsch

### Shanghai Jiao Tong University
- Ze-Guang Han
- Xianbin Su

]

---

background-image: url(assets/USydLogo-black.svg)
background-size: 260px
background-position: 90% 95%

## Thanks!


.pull-left[
.pull-bottom[
### More information
<a href="mailto:yingxin.lin@sydney.edu.a">
.black[`r icon::fa("paper-plane")` yingxin.lin@sydney.edu.au]
</a>

<a href="https://sydneybiox.github.io/scMerge/">
.black[`r icon::fa("link")` sydneybiox.github.io/scMerge/]
</a>

<a href="http://twitter.com/garthtarr">
.black[`r icon::fa("twitter")` @LinYingxin]
</a>

<a href="http://github.com/garthtarr">
.black[`r icon::fa("github")` @yingxinlin]
</a>
]
]