<!DOCTYPE html>
<html>
  <head>
    <title></title>
    <meta charset="utf-8">
    <link href="index_files/remark-css-0.0.1/default.css" rel="stylesheet" />
    <link href="index_files/font-awesome-5.3.1/css/fontawesome-all.min.css" rel="stylesheet" />
    <script type="text/x-mathjax-config">
      MathJax.Hub.Config({
        TeX: { equationNumbers: { autoNumber: "AMS" } },
      });
    </script>
    <style>
    .mjx-mrow a {
      color: black;
      pointer-events: none;
      cursor: default;
    }
    </style>
    <link rel="stylesheet" href="assets/sydney-fonts.css" type="text/css" />
    <link rel="stylesheet" href="assets/sydney.css" type="text/css" />
  </head>
  <body>
    <textarea id="source">






class: title-slide
background-image: url("assets/USydLogo-black.svg"), url("assets/title-image3.jpg")
background-position: 10% 90%, 100% 50%
background-size: 160px, 50% 100%
background-color: #E64626

.pull-left[
# .black[scMerge]

## Integration of multiple single-cell RNA-seq datasets leveraging stable expression and pseudo-replication

### Yingxin Lin

### Nov 30, 2018



&lt;br&gt;&lt;br&gt;&lt;br&gt;

]
---

class: segue-red

.black[# Main challenge in integration of scRNA-seq data: strong batch effect.]

---
class: segue-red

# scMerge algorithm




---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide8.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide9.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide10.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide11.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide12.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide13.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide14.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide15.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide16.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide17.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide18.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide19.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide20.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide21.png")
background-size: contain
background-position: 50% 50%

---
class: segue-red

# Evaluation




---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide31.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/sil.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/sil2.png")
background-size: contain
background-position: 50% 50%

---

class: middle, bottom
background-image: url("assets/scMergeSlides/Slide32.png")
background-size: contain
background-position: 50% 50%

---



## Installing <i class="fas  fa-download fa-pull-right "></i>

The **scMerge** package is in GitHub code repository [<i class="fab  fa-github "></i> `SydneyBioX/scMerge`](https://github.com/SydneyBioX/scMerge)


```r
# Installing scMerge and the data files using
devtools::install_github("SydneyBioX/scMerge.data")
devtools::install_github("SydneyBioX/scMerge")
```

We designed our package to be consistent with the popular BioConductorâ€™s single cell analysis framework, namely the packages `SingleCellExperiment` and `scater`.



```r
library(scMerge.data)
library(scMerge)
library(SingleCellExperiment)
library(scater)
```


---


class: segue-red


# Case Study

Unsupervised scMerge

Supervised scMerge

Semi-supervised scMerge

---

## ESC data

We will use a mouse embryonic stem cell (mESC, E-MTAB-2600) data as an example to illustrate  `scMerge` function. 

.scroll-output[

```r
## mouse ESC data
data("sce_mESC", package = "scMerge.data")
sce_mESC
```

```
## class: SingleCellExperiment 
## dim: 24224 704 
## metadata(0):
## assays(2): counts logcounts
## rownames(24224): ENSMUSG00000000001 ENSMUSG00000000028 ...
##   ENSMUSG00000060565 ENSMUSG00000080069
## rowData names(0):
## colnames(704): ola_mES_lif_1_1.counts ola_mES_lif_1_10.counts ...
##   ola_mES_2i_5_95.counts ola_mES_2i_5_96.counts
## colData names(2): cellTypes batch
## reducedDimNames(0):
## spikeNames(0):
```
]

---

## ESC data

The mESC data contains 5 different batches from three different cell types. Using a PCA plot, we can see that despite strong separation of cell types, there is also a strong separation due to batch effects. 


.pull-left[

```r
table(sce_mESC$batch, sce_mESC$cellTypes)
```

```
##         
##          2i a2i lif
##   batch1  0   0  81
##   batch2 82  93  90
##   batch3 59  66  79
##   batch4 72   0   0
##   batch5 82   0   0
```
]

.pull-right[

```r
scater::plotPCA(sce_mESC, 
                colour_by = "cellTypes", 
                shape_by = "batch")
```

![](index_files/figure-html/unnamed-chunk-5-1.png)&lt;!-- --&gt;
]



---
## ESC data: running unsupervised `scMerge`



Key inputs of `scMerge`:

- `sce_combine`: a `SingleCellExperiment` object contains the batch-combined matrix with batch info in `colData`.
- `ctl`: a gene list of negative control genes.
- `assay_name`: The assay name for the adjusted expression matrix. 

--

In unsupervised scMerge, we will perform a k-means clustering to obtain pseudo-replicates. This requires the users to supply a `kmeansK` vector with each element indicating number of clusters in each of the batches. 

--

For example, we know the first, forth, and fifth batch contain only one cell type and the second and third batch contain three cell types. Hence, `kmeansK = c(1,3,3,1,1)` in this case.

--


```r
sce_mESC &lt;- scMerge(sce_combine = sce_mESC, 
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_unsupervised")
```

---
## ESC data: running unsupervised `scMerge`


```r
sce_mESC &lt;- scMerge(sce_combine = sce_mESC, 
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_unsupervised")
```

The normalised data matrix will be stored in slot `assays` with the list name as `"scMerge_unsupervised"` of `sce_mESC`.

--

.blockquote[
### <i class="fas  fa-comment-dots "></i> Comments

scMerge is robust to overestimation of number of clusters in general (See our manuscript for details).
]


---

## ESC data: running unsupervised `scMerge`

We now construct the PCA plot again on our normalised data. We can observe a much better separation by cell type and less separation by batches.


```r
scater::plotPCA(sce_mESC, 
                colour_by = "cellTypes", 
                shape_by = "batch",
                run_args = list(exprs = "scMerge_unsupervised"))
```

![](index_files/figure-html/unnamed-chunk-8-1.png)&lt;!-- --&gt;


---

## Other diagnostic plots

**Explanatory variables plot**: density plot of percentage of phenotypic variance explained.






.pull-left[

```r
scater::plotExplanatoryVariables(sce_mESC,
        method = "density",
        exprs_values = "logcounts", 
        variables = c("batch", "cellTypes",
                      "total_counts",
                      "total_features")) + 
  labs(title = "logcounts")
```

![](index_files/figure-html/unnamed-chunk-10-1.png)&lt;!-- --&gt;
]

.pull-right[

```r
scater::plotExplanatoryVariables(sce_mESC,
        method = "density",
        exprs_values = "scMerge_unsupervised", 
        variables = c("batch", "cellTypes",
                      "total_counts",
                      "total_features")) + 
  labs(title = "scMerge")
```

![](index_files/figure-html/unnamed-chunk-11-1.png)&lt;!-- --&gt;
]

---

## ESC data: running supervised `scMerge`

If **all** cell type information is available to the users, or users has already performed clustering analysis using their favourite method, then it is possible to use this information to create pseudo-replicates. 

This can be done through the `cell_type` argument in the `scMerge` function.



```r
sce_mESC &lt;- scMerge(sce_mESC,
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_supervised",
*                  cell_type = sce_mESC$cellTypes)
```

---

## ESC data: running semi-supervised `scMerge` 

If the user is only able to access **partial** cell type information (or have confidence in only partial cell type informaton), then it is still possible to use this information to create pseudo-replicates. 

This can be done through the `cell_type` and `cell_type_inc` arguments in the scMerge function. `cell_type_inc` should contain a vector of indices indicating which elements in the `cell_type` vector should be used to perform semi-supervised `scMerge`.



```r
sce_mESC &lt;- scMerge(sce_mESC,
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_semisupervised1",
                    cell_type = sce_mESC$cellTypes,
*                  cell_type_inc = which(sce_mESC$cellTypes == "2i"))
```

---


## ESC data: running semi-supervised `scMerge` 
There is alternative semi-supervised method to create pseudo-replicates for scMerge. This uses known cell type information to identify mutual nearest clusters.
It is achieved via the `cell_type` and `cell_type_match = TRUE` options in the `scMerge` function.



```r
sce_mESC &lt;- scMerge(sce_mESC,
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_semisupervised2",
                    cell_type = sce_mESC$cellTypes,
*                    cell_type_match = TRUE)
```


---

## Achieving fast computation: `fastRUVIII` <i class="fas  fa-rocket fa-pull-right "></i>


`RUVIII` function in the package `ruv` is great but slow for the large single-cell data. We develop a fast version of `RUVIII`, by using 
- *randomised SVD algorithm*: to obtain a very accurate approximation of the noise structure in the data by performing a partial SVD. 
-  package `Rcpp` to speed up the matrix calculation.
- `rsvd_prop` is a parameter between 0 and 1, controlling the degree of approximations.

.pull-left[

```r
fastRUVIII(Y, M, ctl, k = NULL, fast_svd = TRUE,
           rsvd_prop = 0.1)
```


```r
sce_mESC &lt;- scMerge(sce_mESC, 
                    ctl = segList_ensemblGeneID$mouse$mouse_scSEG,
                    kmeansK = c(1,3,3,1,1),
                    assay_name = "scMerge_fast", 
*                   fast_svd = TRUE, 
*                   rsvd_prop = 0.1)
```

]

--

.pull-right[
.blockquote[
### <i class="fas  fa-comment-dots "></i> Comments

We recommend only use this option when the number of cells is large in your single cell data. 
]]


---


background-image: url(assets/scMergeSlides/speedup.png)
background-size: 550px
background-position: 20% 80%

## Achieving fast computation: `fastRUVIII` <i class="fas  fa-rocket fa-pull-right "></i>

For the case with 10k cell, the speed up of `fastRUVIII` is about 10 times, compared to `RUVIII` from package `ruv`. 


---

## More on stably expressed genes

### Other negative controls

`scMerge` also provides some other gene lists of negative control genes.


```r
data("segList")
str(segList)
```

```
## List of 2
##  $ human:List of 3
##   ..$ human_scSEG     : chr [1:1076] "AAR2" "AATF" "ABCF3" "ABHD2" ...
##   ..$ bulkMicroarrayHK: chr [1:553] "AATF" "ABL1" "ACAT2" "ACTB" ...
##   ..$ bulkRNAseqHK    : chr [1:3803] "AAGAB" "AAMP" "AAR2" "AARS" ...
##  $ mouse:List of 3
##   ..$ mouse_scSEG     : chr [1:826] "Nfatc3" "Ythdf2" "Gtf2e2" "Mrpl45" ...
##   ..$ bulkMicroarrayHK: chr [1:553] "Aatf" "Abl1" "Acat2" "Actb" ...
##   ..$ bulkRNAseqHK    : chr [1:3803] "Aagab" "Aamp" "Aar2" "Aars" ...
```

---

## More on stably expressed genes


### Alternatively, you define your own stably expressed gene list

`scSEGIndex()` will calculate the single-cell stably expressed gene index to identify stably expressed genes (SEGs) from a log transformed single-cell data. The data here is assumed to be no batch effect and covered a wide range of cell types.


```r
scSEGIndex(exprsMat, cell_type = NULL, ncore = 1)
```


---

## Summary

- A novel algorithm to remove unwanted variation from multiple single cell RNA-seq datasets, allowing for multiple experimental set ups.  This is achieved by identifying stably expressed genes and estimating pseudo replication between data sets.
- Analysis of many publicly available datasets reveals that our algorithm performs well in general
- Semi-supervised and supervised version of scMerge allows people to incoporate prior knowledge faciliated by experimental design.



---

## Reference list





Ghazanfar, S, A. J. Bisogni, J. T. Ormerod, et al. (2016).
"Integrated single cell data analysis reveals cell specific
networks and novel coactivation markers". In: _BMC systems
biology_ 10.5, p. 127.

Lin, Y, S. Ghazanfar, D. Strbenac, et al. (2017). "Evaluating
stably expressed genes in single cells". In: _bioRxiv_, p. 229815.

Lin, Y, S. Ghazanfar, K. Wang, et al. (2018). "scMerge:
Integration of multiple single-cell transcriptomics datasets
leveraging stable expression and pseudo-replication". In:
_bioRxiv_, p. 393280.



---

background-image: url(assets/USydLogo-black.svg)
background-size: 260px
background-position: 5% 95%

## Acknowledgement


.pull-left[
### The University of Sydney
- Jean Yang
- Pengyi Yang
- John Ormerod
- Shila Ghazanfar
- Kevin Wang
- Kitty Lo
]


.pull-right[
### WEHI
- Terry Speed

### The University of Michigan
- Johann Gagnon-Bartsch

### Shanghai Jiao Tong University
- Ze-Guang Han
- Xianbin Su

]

---

background-image: url(assets/USydLogo-black.svg)
background-size: 260px
background-position: 90% 95%

## Thanks!


.pull-left[
.pull-bottom[
### More information

&lt;a href="https://sydneybiox.github.io/scMerge/"&gt;
.black[<i class="fas  fa-link "></i> sydneybiox.github.io/scMerge/]
&lt;/a&gt;

&lt;a href="http://twitter.com/garthtarr"&gt;
.black[<i class="fab  fa-twitter "></i> @LinYingxin]
&lt;/a&gt;

&lt;a href="http://github.com/garthtarr"&gt;
.black[<i class="fab  fa-github "></i> @yingxinlin]
&lt;/a&gt;
]
]
    </textarea>
<script src="https://remarkjs.com/downloads/remark-latest.min.js"></script>
<script src="assets/remark-zoom.js"></script>
<script src="https://platform.twitter.com/widgets.js"></script>
<script>var slideshow = remark.create({
"highlightStyle": "github",
"highlightLines": true,
"countIncrementalSlides": false,
"ratio": "16:9",
"navigation": {
"scroll": false
}
});
if (window.HTMLWidgets) slideshow.on('afterShowSlide', function (slide) {
  window.dispatchEvent(new Event('resize'));
});
(function() {
  var d = document, s = d.createElement("style"), r = d.querySelector(".remark-slide-scaler");
  if (!r) return;
  s.type = "text/css"; s.innerHTML = "@page {size: " + r.style.width + " " + r.style.height +"; }";
  d.head.appendChild(s);
})();</script>

<script>
(function() {
  var i, text, code, codes = document.getElementsByTagName('code');
  for (i = 0; i < codes.length;) {
    code = codes[i];
    if (code.parentNode.tagName !== 'PRE' && code.childElementCount === 0) {
      text = code.textContent;
      if (/^\\\((.|\s)+\\\)$/.test(text) || /^\\\[(.|\s)+\\\]$/.test(text) ||
          /^\$\$(.|\s)+\$\$$/.test(text) ||
          /^\\begin\{([^}]+)\}(.|\s)+\\end\{[^}]+\}$/.test(text)) {
        code.outerHTML = code.innerHTML;  // remove <code></code>
        continue;
      }
    }
    i++;
  }
})();
</script>
<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
(function () {
  var script = document.createElement('script');
  script.type = 'text/javascript';
  script.src  = 'https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-MML-AM_CHTML';
  if (location.protocol !== 'file:' && /^https?:/.test(script.src))
    script.src  = script.src.replace(/^https?:/, '');
  document.getElementsByTagName('head')[0].appendChild(script);
})();
</script>
  </body>
</html>
